addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from collections import defaultdict
+
+from odoo import _, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class MailTrackingDurationMixin(models.AbstractModel):
+    _name = "mail.tracking.duration.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
```

Comment:
I tried to find a more generic description without involving "stage", but did not find. But if you feel inspired ...